src/explorers/AAPS.jl

```diff
@@ -0,0 +1,243 @@
+###############################################################################
+# The Pigeons implementation of AAPS is based on code by 
+# Naitong Chen and Trevor Campbell (2023). Reused with their permission.
+# Note:
+# y(x) := M^{-1/2}x => x(y) = M^{1/2}y
+# x ~ p_x => p_y(y) = p_x(M^{1/2}y) |detM|^{1/2} propto p_x(M^{1/2}y) 
+# => grad{log p_y}(y) = M^{1/2} grad(log p_x)(M^{1/2}y) 
+# for p ~ N(0,M), the leapfrog is
+#   p*(x,p)   = p + (eps/2)grad(x)
+#   x'(x,p*)  = x + epsM^{-1}p*
+#   p'(x',p*) = p* + (eps/2)grad(x')
+# if y = M^{-1/2}p => y ~ N(0,I) and p = M^{1/2}y.
+#   y*(x,y)   = M^{1/2}y + (eps/2)grad(x)
+#   x'(x,y*)  = x + epsM^{-1}y*
+#   y'(x',y*) = y* + (eps/2)grad(x')
+###############################################################################
+
+""" 
+$SIGNATURES 
+
+The Apogee to Apogee Path Sampler (AAPS) by Sherlock et al. (2022). 
+
+AAPS is a simple alternative to the No U-Turn Sampler (NUTS).
+It serves a similar purpose as NUTS: the method should be robust to its choice 
+of tuning parameters when compared to standard HMC.
+For a given starting position and momentum (x, v), AAPS explores both forward and 
+backward trajectories. The trajectories are divided into segments, with 
+segments being separated by apogees (local maxima) in the energy landscape 
+of -log pi(x). The tuning parameter `K` defines the number of segments to explore. 
+"""
+Base.@kwdef struct AAPS{T,D}
+    """ 
+    Leapfrog step size.
+    """
+    ϵ::Float64 = 1.0
```

Comment:
I suggest we use names more aligned with the AutoMALA struct, e.g. `step_size`, some descriptive name instead of K, etc

Reply:
on it!